### PR TITLE
refactor(web): replace .e-icon with <e-icon>

### PR DESCRIPTION
**Pull request by TrygveUrdahl**
# PR Checklist
https://elvia.atlassian.net/wiki/spaces/TEAMATOM/pages/10427498683/Review+prosess

- [ ] Bumpet package?
- [ ] Updated changelog?
- [ ] Correct date in changelog?

### CODE REVIEW

File: packages/web/src/app/doc-pages/brand/icon-doc/icon-generator/icon-generator.component.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  standalone: true,
+  imports: [],
+  template: `<e-icon [name]="transformedName()" [size]="size()" />`,
+  styles: [':host { line-height: 0; }'],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+})
+export class IconGeneratorComponent {
+  name = input.required<string | undefined>();
+  size = input<'sm' | 'md'>();
+  transformedName = computed(() => kebabCaseToCamelCase(this.name() ?? ''));
```
Comment (TomSchrier) [inline packages/web/src/app/doc-pages/brand/icon-doc/icon-generator/icon-generator.component.ts:16]: Neat💡

File: packages/web/src/app/shared/component-documentation/ceg/controls/counter/counter.component.html [hunk 2]
```diff
@@ -7,12 +7,8 @@
...
   >
     <span class="e-btn__icon">
-      <i
-        class="e-icon e-icon--subtract_circle"
-        [ngClass]="{ 'e-icon--color-disabled': counter.value <= counter.min || disabled }"
-        aria-hidden="true"
-      ></i>
-      <i class="e-icon e-icon e-icon--subtract_circle-filled-color" aria-hidden="true"></i>
+      <e-icon name="subtractCircle" />
+      <e-icon name="subtractCircleFilledColor" />
```
Comment (TomSchrier) [inline packages/web/src/app/shared/component-documentation/ceg/controls/counter/counter.component.html:11]: Is this one missing `[ngClass]="{ 'e-icon--color-disabled': counter.value <= counter.min || disabled }"`?

File: packages/web/src/app/shared/component-documentation/ceg/controls/counter/counter.component.html [hunk 3]
```diff
@@ -27,12 +23,8 @@
...
   >
     <span class="e-btn__icon">
-      <i
-        class="e-icon e-icon--add_circle"
-        [ngClass]="{ 'e-icon--color-disabled': counter.value >= counter.max || disabled }"
-        aria-hidden="true"
-      ></i>
-      <i class="e-icon e-icon e-icon--add_circle-filled-color" aria-hidden="true"></i>
+      <e-icon name="addCircle" />
+      <e-icon name="addCircleFilledColor" />
```
Comment (TomSchrier) [inline packages/web/src/app/shared/component-documentation/ceg/controls/counter/counter.component.html:27]: Ditto

File: packages/web/src/app/doc-pages/brand/icon-doc/icon-generator/icon-generator.component.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { CUSTOM_ELEMENTS_SCHEMA, Component, computed, input } from '@angular/core';
+
+import { kebabCaseToCamelCase } from '../icon-preview/utils';
+
+@Component({
+  selector: 'app-icon-generator',
+  standalone: true,
+  imports: [],
```
Comment (TomSchrier) [inline packages/web/src/app/doc-pages/brand/icon-doc/icon-generator/icon-generator.component.ts:8]:
```suggestion
```

**Comment by TomSchrier** [comment]
Something is up with the Search field for the changelog, but this is prod too, so i wont blame you 🤷🏽
Skjermbilde 2024-08-30 kl  08 11 13
Skjermbilde 2024-08-30 kl  08 16 08

File: packages/web/src/app/shared/component-documentation/ceg/controls/counter/counter.component.html (hunk 2 above)
Comment (TrygveUrdahl) [reply packages/web/src/app/shared/component-documentation/ceg/controls/counter/counter.component.html:11]: Nope, it doesn't need it! The icon color propagates from `[disabled]` on the buttons!

**Review by TomSchrier — APPROVED** [review]
LGTM 🎉

**Pull request merged**